scalafix-rules/src/main/scala/scalafix/internal/rule/OrganizeImports.scala

```diff
@@ -707,32 +719,40 @@ object OrganizeImports {
       scalacOptions: List[String],
       scalaVersion: String
   ): Configured[Rule] = {
-    val hasCompilerSupport = scalaVersion.startsWith("2")
-
-    val hasWarnUnused = hasCompilerSupport && {
-      val warnUnusedPrefix = Set("-Wunused", "-Ywarn-unused")
-      val warnUnusedString = Set("-Xlint", "-Xlint:unused")
-      scalacOptions exists { option =>
-        (warnUnusedPrefix exists option.startsWith) || (warnUnusedString contains option)
-      }
-    }
+    ScalaVersion.from(scalaVersion).map { v =>
+        v.isScala2 || (
+          v.isScala3 &&
+            v.minor.getOrElse(0) >= 3 &&
+            v.patch.getOrElse(0) >= 1
+        )
+    } match {
+      case Failure(exception) => Configured.error(exception.getMessage())
+      case Success(hasCompilerSupport) =>
+        val hasWarnUnused = hasCompilerSupport && {
+          val warnUnusedPrefix = Set("-Wunused", "-Ywarn-unused")
+          val warnUnusedString = Set("-Xlint", "-Xlint:unused")
+          scalacOptions exists { option =>
+            (warnUnusedPrefix exists option.startsWith) || (warnUnusedString contains option)
+          }
+        }
 
-    if (!conf.removeUnused || hasWarnUnused)
-      Configured.ok(new OrganizeImports(conf))
-    else if (hasCompilerSupport)
-      Configured.error(
-        "The Scala compiler option \"-Ywarn-unused\" is required to use OrganizeImports with"
-          + " \"OrganizeImports.removeUnused\" set to true. To fix this problem, update your"
-          + " build to use at least one Scala compiler option like -Ywarn-unused-import (2.11"
-          + " only), -Ywarn-unused, -Xlint:unused (2.12.2 or above) or -Wunused (2.13 only)."
-      )
-    else
-      Configured.error(
-        "\"OrganizeImports.removeUnused\" is not supported on Scala 3 as the compiler is"
-          + " not providing enough information. Run the rule with"
-          + " \"OrganizeImports.removeUnused\" set to false to organize imports while keeping"
-          + " potentially unused imports."
-      )
+        if (!conf.removeUnused || hasWarnUnused)
+          Configured.ok(new OrganizeImports(conf))
+        else if (hasCompilerSupport)
+          Configured.error(
+            "The Scala compiler option \"-Ywarn-unused\" is required to use OrganizeImports with"
+              + " \"OrganizeImports.removeUnused\" set to true. To fix this problem, update your"
+              + " build to use at least one Scala compiler option like -Ywarn-unused-import (2.11"
+              + " only), -Ywarn-unused, -Xlint:unused (2.12.2 or above) or -Wunused (2.13 only)."
+          )
+        else
+          Configured.error(
+            "\"OrganizeImports.removeUnused\"" + s"is not supported on $scalaVersion as the compiler is"
+              + " not providing enough information. Please upgrade Scala 3 version to 3.3.1 or greater."
```

Comment:
he version needs to be aligned with the check above - either refactor to use common val, or keep this comment open to double=check at merge time